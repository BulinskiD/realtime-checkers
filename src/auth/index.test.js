import React from 'react';
import {shallow} from 'enzyme';
import {Login} from './index';
import {act} from 'react-dom/test-utils';
const loginWithEmailAndPassword = jest.fn();
const history = {};
const preventDefault = jest.fn();
const formSubmit = {preventDefault};

describe("Auth", () => {
   it("should render correctly with given props", () => {
      const LoginComp = shallow(<Login loginWithEmailAndPassword={loginWithEmailAndPassword} history={history} />);
      expect(LoginComp).toMatchSnapshot();
   });

   it("should handle login correctly", () => {
      const email = {target: {value: null}};
      email.target.value = "email";

      const password = {target: {value: null}};
      password.target.value = "password";

      const LoginComp = shallow(<Login loginWithEmailAndPassword={loginWithEmailAndPassword} history={history} />);

      act(() => {
         LoginComp.find('.email').props().onChange(email);
      });

      act(() => {
         LoginComp.find('.password').props().onChange(password);
      });

      act(() => {
         LoginComp.find('Form').props().onSubmit(formSubmit);
      });

      expect(preventDefault).toHaveBeenCalledTimes(1);
      expect(loginWithEmailAndPassword).toHaveBeenCalledWith('email', 'password', history);
   });
});