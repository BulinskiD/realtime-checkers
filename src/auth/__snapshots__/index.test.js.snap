// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should render correctly with given props 1`] = `
<div
  className="d-flex justify-content-center align-items-center"
>
  <Form
    className="w-25 mt-4"
    inline={false}
    onSubmit={[Function]}
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Email
    </FormLabel>
    <FormControl
      aria-label="email"
      className="mb-4 email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
    <FormLabel
      column={false}
      srOnly={false}
    >
      Has≈Ço
    </FormLabel>
    <FormControl
      className="mb-4 password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <Button
      active={false}
      disabled={false}
      type="submit"
      variant="primary"
    >
      Zaloguj
    </Button>
  </Form>
</div>
`;
